[docs][maps] Add custom marker icon examples (#47296)
expo/expo: docs/pages/versions/unversioned/sdk/maps.mdx

}
```

### Custom marker icons

> Use `useImage` from `expo-image` to load custom marker and annotation icons.

Reviewer: ```suggestion
You can use `useImage` hook from `expo-image` to load custom marker and annotation icons.
```

Please add a link to the `useImage` definition.

<Tabs>
<Tab label="Google Maps">

> 

Reviewer: I'd suggest adding a one line description here and in Apple Maps tab, as to, what the following example is about.

```tsx
import { useImage } from 'expo-image';
import { GoogleMaps } from 'expo-maps';

export default function Map() {
  const icon = useImage('https://example.com/marker.svg', { maxWidth: 48, maxHeight: 48 });

  return (
    <GoogleMaps.View
      style={{ flex: 1 }}
      markers={[
        {
          coordinates: { latitude: 37.78825, longitude: -122.4324 },
          icon: icon ?? undefined,
          anchor: { x: 0.5, y: 0.5 },
        },
      ]}
    />
  );
}
```

> `GoogleMaps.Marker.icon` expects an image reference, such as the value returned by the `useImage` hook from the `expo-image` package. It does not accept an image source directly. The loaded image dimensions determine the marker size, not a marker style prop. For SVG icons, set `width`, `height`, and `viewBox` in the SVG, or pass `maxWidth` and `maxHeight` to `useImage`. Use `anchor` to align custom icons with their coordinates.

Reviewer: ```suggestion
`GoogleMaps.Marker.icon` expects an image reference, such as the value returned by the `useImage` hook from the `expo-image` package. It does not accept an image source directly. 

The loaded image dimensions determine the marker size, not a marker style prop. For SVG icons, set `width`, `height`, and `viewBox` in the SVG, or pass `maxWidth` and `maxHeight` to `useImage` hook. You can use `anchor` to align custom icons with their coordinates.
```

</Tab>
<Tab label="Apple Maps">

```tsx
import { useImage } from 'expo-image';
import { AppleMaps } from 'expo-maps';

export default function Map() {
  const icon = useImage('https://example.com/marker.svg');

  return (
    <AppleMaps.View
      style={{ flex: 1 }}
      annotations={[
        {
          coordinates: { latitude: 37.78825, longitude: -122.4324 },
          icon: icon ?? undefined,
        },
      ]}
    />
  );
}
```

> `AppleMaps.Annotation.icon` expects an image reference, such as the value returned by the `useImage` hook from the `expo-image` package. Use `AppleMaps.Annotation` for custom image icons. `AppleMaps.Marker` supports marker-specific options such as `systemImage`, `monogram`, and `tintColor`.

Reviewer: ```suggestion
`AppleMaps.Annotation.icon` expects an image reference, such as the value returned by the `useImage` hook from the `expo-image` library. You can use `AppleMaps.Annotation` for custom image icons. `AppleMaps.Marker` supports marker-specific options such as `systemImage`, `monogram`, and `tintColor`.
```
